crates/sui-protocol-config-macros/src/lib.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+extern crate proc_macro;
+
+use proc_macro::TokenStream;
+use quote::{format_ident, quote};
+use syn::{parse_macro_input, Data, DeriveInput, Fields, Type};
+
+/// This macro derives getters for each field in ProtocolConfig at a specific version
+/// If the field is empty, None is returned, otherwise Some(value) is returned
+#[proc_macro_derive(ProtocolConfigGetters)]
```

Comment:
Curious why you made a separate crate for this, instead of putting it in sui-protocol-conifg? Is that a requirement for proc macros or something? (I haven't worked with them a lot)

Reply:
Its a requirement for proc-macros.
Same reason `mysten-util-mem-derive` and `sui-open-rpc-macros` are in their own crates